src/resolve-config.js

```diff
@@ -19,23 +21,43 @@ function getLoadFunction(opts) {
   return getExplorerMemoized(opts).load;
 }
 
-function resolveConfig(filePath, opts) {
+function _resolveConfig(filePath, opts, sync) {
   opts = Object.assign({ useCache: true }, opts);
-  const load = getLoadFunction({ cache: !!opts.useCache, sync: false });
-  return load(filePath, opts.config).then(result => {
-    return !result ? null : mergeOverrides(result.config, filePath);
-  });
+  const loadOpts = { cache: !!opts.useCache, sync: !!sync };
+  const load = getLoadFunction(loadOpts);
+  const loadEditorConfig = resolveEditorConfig.getLoadFunction(loadOpts);
+  const arr = [load, loadEditorConfig].map(l => l(filePath, opts.config));
+
+  const unwrapAndMerge = arr => {
+    const result = arr[0];
+    const editorConfigged = arr[1];
```

Comment:
typo?

Reply:
I had intended for this to be shorthand for `editorConfigured`, but it's not exactly clear, so I changed it in https://github.com/prettier/prettier/pull/2760/commits/ccc3b0151b388c5a717e9c5065b04352e125b20d